docroot/modules/custom/mymodule/src/Form/CustomSettingsForm.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Drupal\mymodule\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Configure mymodule settings for this site.
+ */
+class CustomSettingsForm extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'mymodule_custom_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return ['mymodule.settings'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('mymodule.settings');
+    $form['custom_name'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Custom Name'),
+      '#default_value' => $config->get('custom_name'),
+    ];
+    $form['custom_checkbox'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('This is a checkbox example.'),
+      '#default_value' => $config->get('custom_checkbox'),
+    ];
+    $form['actions']['#type'] = 'actions';
+    $form['actions']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Save Custom Configs'),
+      '#button_type' => 'primary',
+    ];
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+
+    parent::validateForm($form, $form_state);
+  }
```

Comment:
As this is optional and no validation is here, no need to add it here.

Reply:
It is here for further development, I had a validation line to test this method.
I removed that validation to add something more reasonable. I'll be adding that in future implementations.